metagpt/utils/resp_parse.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# @Date    : 2023/8/22 22:18
+# @Author  : stellahong (stellahong@fuzhi.ai)
+# @Desc    :
+import json5
+import re
+
+
+def flatten_json_structure(json_array):
+    if (isinstance(json_array, list) and len(json_array) == 1 and not isinstance(json_array[0], str)):
+        return flatten_json_structure(json_array[0])
+    
+    if (isinstance(json_array, dict) and len(json_array.values()) == 1 and not isinstance(list(json_array.values())[0],
+                                                                                          str)):
+        return flatten_json_structure(list(json_array.values())[0])
+    
+    flattened_json_array = []
+    
+    if (isinstance(json_array, dict)):
+        json_array = json_array.values()
+    
+    for json_object in json_array:
+        flattened_dict = flatten_json_object(json_object)
+        flattened_values = ", ".join(str(v) for v in flattened_dict.values())
+        flattened_json_array.append(flattened_values)
+    
+    return flattened_json_array
+
+
+def flatten_json_object(obj, parent_key='', sep=', '):
+    if isinstance(obj, str):
+        return dict([("value", obj)])
+    
+    if isinstance(obj, list):
+        return dict([("value", sep.join(str(v) for v in obj))])
+    
+    items = []
+    for key, value in obj.items():
+        new_key = f"{parent_key}{sep}{key}" if parent_key else key
+        if isinstance(value, dict):
+            items.extend(flatten_json_object(value, new_key, sep=sep).items())
+        elif isinstance(value, list):
+            items.append((new_key, sep.join(str(v) for v in value)))
+        else:
+            items.append((new_key, value))
+    return dict(items)
+
+
+def try_parse_json(input_text):
```

Comment:
can you provide some `input_text` examples or corresponding UT to make the code more clear. It contains some nested call and boundary treatment.